tests/api/api-resources.test.js

```diff
@@ -0,0 +1,272 @@
+// Copyright (c) 2020 Red Hat, Inc.
+
+jest.retryTimes(global.retry)
+
+const { exec, execSync } = require('child_process')
+const lodash = require('lodash')
+
+const squad = require('../../config').get('squadName')
+const {
+  getKubeConfig,
+  getSearchApiRoute,
+  getToken,
+  searchQueryBuilder,
+  sendRequest,
+} = require('../common-lib/clusterAccess')
+
+// Generate list of resources to be tested.
+const testList = ['configmap', 'deployment', 'pod', 'secret']
+
+// Error
+const REQUESTED_RESOURCE_NOT_FOUND_ERROR =
+  'server could not find the requested resource'
+
+/**
+ * Base test for kubernetes
+ * @param {*} kind
+ * @param {*} apigroup
+ * @param {*} cluster
+ * @param {*} user
+ * @param {*} namespace
+ */
+function baseTest(
+  kind,
+  apigroup = { name, useAPIGroup: false },
+  cluster = { type: 'hub', name: 'local-cluster' },
+  user = 'kubeadmin',
+  namespace = '--all-namespaces'
+) {
+  var runTest = test
+  var property = kind
+  var res = []
+
+  // Check to see if the test needs to include the apigroup name within the query. For kind resources with v1 versions, no apigroup is needed.
+  if (apigroup.useAPIGroup && apigroup.name != 'v1')
+    property += `.${apigroup.name}`
+
+  var cmd = `oc get ${property.toLowerCase()} ${
+    namespace === '--all-namespaces' ? namespace : `-n ${namespace}`
+  } --no-headers `
+
+  // We need to exclude managed cluster results from out request.
+  if (cluster.type === 'hub' && process.env.OPTIONS_MANAGED_CLUSTER_NAME)
+    cmd += `--field-selector='metadata.namespace!=${process.env.OPTIONS_MANAGED_CLUSTER_NAME}'`
+
+  // Uncomment the following line for debugging purposes.
+  console.debug(cmd)
+
+  try {
+    res = removeEmptyEntries(
+      execSync(cmd, { stdio: [] }).toLocaleString().split('\n')
+    )
+  } catch (err) {
+    runTest = test.skip
+  }
+
+  runTest(`[P2][Sev2][${squad}] verify data integrity for resource property: ${
+    apigroup.useAPIGroup ? kind : `${kind}.${apigroup.name}`
+  }`, async () => {
+    const filters = [
+      { property: 'cluster', values: [cluster.name] },
+      { property: 'kind', values: [kind] },
```

Comment:
Revert the order of this query for better performance.
```suggestion
      { property: 'kind', values: [kind] },
      { property: 'cluster', values: [cluster.name] },
```

Each resource needs to pass all filters, so the sooner the query eliminates resources, the less work.

Reply:
Actually, I'd also move the conditionals so you end with this order.
```
namespace -> kind -> group -> cluster
```